lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -463,8 +463,8 @@ def __str__(self):
     def __unicode__(self):
         if self.fqcn is None:
             return (
-                u'"virtual collection Git repo"' if self.is_scm
-                else u'"virtual collection namespace"'
+                f'"{self.type} collection Git repo"' if self.is_scm
+                else f'"{self.type} collection namespace"'
```

Comment:
So we may have to do something with those double quotes, as they'll probably make the error messages weird. Would removing them work?
```suggestion
                f'{self.type} collection from a Git repo' if self.is_scm
                else f'{self.type} collection from a namespace'
```